pkg/controller/atlasproject/atlasproject_controller.go

```diff
@@ -94,18 +98,21 @@ func (r *AtlasProjectReconciler) Delete(obj runtime.Object) error {
 }
 
 func (r *AtlasProjectReconciler) SetupWithManager(mgr ctrl.Manager) error {
-	return ctrl.NewControllerManagedBy(mgr).
-		// Listen to all kiinds
-		For(&mdbv1.AtlasProject{}).
-		Watches(
-			&source.Kind{Type: &mdbv1.AtlasProject{}},
-			&watch.DeleteEventHandler{Controller: r},
-			builder.WithPredicates(watch.DeleteOnly()),
-		).
-		Watches(
-			&source.Kind{Type: &corev1.Secret{}},
-			&watch.ResourcesHandler{TrackedResources: r.WatchedResources},
-		).
-		WithEventFilter(watch.CommonPredicates()).
-		Complete(r)
+	c, err := controller.New("AtlasProject", mgr, controller.Options{Reconciler: r})
```

Comment:
the builder is awful, I fought with it but it just resulted in duplicated handlings....
So I did everything directly
the cluster manager so far doesn't need this though